packages/block-editor/src/components/block-list/index.native.js

```diff
@@ -286,13 +291,15 @@ export default compose( [
 		} = select( 'core/block-editor' );
 
 		const selectedBlockClientId = getSelectedBlockClientId();
+		const blockClientIds = getBlockOrder( rootClientId );
 
 		return {
-			blockClientIds: getBlockOrder( rootClientId ),
+			blockClientIds,
 			blockCount: getBlockCount( rootClientId ),
 			getBlockName,
 			isBlockSelected,
 			selectedBlock: getSelectedBlock(),
+			firstBlock: getBlock( blockClientIds[ 0 ] ),
```

Comment:
Should we defend ourselves here if `blockClientIds() ` at some scenario can be null?

Reply:
Currently, we're protected from this because `blockClientIds` is assigned from `getBlockOrder( rootClientId )`, which [is protected to always return at least an empty array](https://github.com/WordPress/gutenberg/blob/master/packages%2Fblock-editor%2Fsrc%2Fstore%2Fselectors.js#L776):

```
export function getBlockOrder( state, rootClientId ) {
	return state.blocks.order[ rootClientId || '' ] || EMPTY_ARRAY;
}
```

With that said, I wouldn't be particularly opposed to still adding in the protection since it doesn't hurt and it would protect us against a change to the implementation of `getBlockOrder`. What do you think?

As an aside, it's cases like this that make me miss working with typescript (or flow). Felt good knowing that the compiler would catch this if it was a problem.

Reply:
Thanks for the details explanation. I think we are fine with the current solution :)